### DISCUSSION TITLE
CDN setup with .xml file

### DISCUSSION THREAD

**Discussion by mahyarwd** [crowdin/mobile-sdk-ios#201] [Q&A] [answered]
Hey guys

I'm trying to implement the SDK in our iOS app, but I'm having difficulty with getting it to work. 
The issue I'm having is that the translation files dont get downloaded at all. After some playing around I found out that the problem seems to be with us using ".xml" instead of ".string" file. 
I couldnt find any documentation on this, so I'm not sure if it's me doing something wrong or the SDK just doesnt support .xml files. 

Is .xml files not support? 

Thanks in advance!

**Comment by andrii-bodnar** [comment]
Hi @mahyarworkday!

The iOS SDK works only with the native iOS strings/stringsdict file formats. Also, the iOS XLIFF is supported

What kind of XML files you are using?

**Comment by mahyarwd** [comment]
Hey @andrii-bodnar 

I see, then it makes a bit more sense. 
The xml file i'm using looks pretty much like this: 

```
<?xml version="1.0" encoding="utf-8"?>
<resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
    <string name="app_name" comment="NOT TO BE TRANSLATED">App Name</string>
    <string name="string_inside_app">Title</string>
```

**Comment by andrii-bodnar** [comment]
@mahyarworkday this file looks like a regular Android XML file

There is a "Bundles" feature in Crowdin. This feature allows adding and configuring bundles for exporting sets of strings in one of the selected formats and this feature supports export to the iOS Strings file format.

Then, bundles can be used in the Content Delivery while creating a Distribution for your iOS app.

Read more - [Bundles](https://support.crowdin.com/advanced-project-setup/#bundles)

**Reply by mahyarwd** [reply]
So do you mean that I need to create a separate `Localizable.strings` and `Localizable.stringsdict` bundle in the `Settings > Target File Bundles`? 
Or do you mean that i should setup a `Add Distribution -> Bundle -> Export Options` and then choose iOS Strings? In the case of adding an Export Option in the Distribution settings, what should the Taget files pattern be in order to receive these files: `Localizable.strings`, `Localizable.stringsdict` and `APPWidget.strings`? 
I have tried setting the distribution up with a few of the options and the only thing that worked was when I set the Target Files Pattern to be `Localizable.strings` but that would then only work for my normal string and not plurals and widget

Thanks in advance!

**Reply by andrii-bodnar** [reply, marked as answer]
You need to configure two [bundles](https://support.crowdin.com/bundles/) in the Crowdin project:

- [iOS Strings](https://store.crowdin.com/ios-string-exporter)
- [Stringsdict](https://store.crowdin.com/stringsdict-export) (in case you have plural strings in your project)

Then create a new distribution in the _Target file bundle_ export mode and select both previously created bundles.

You can do this either in the UI or using the CLI [`bundle`](https://crowdin.github.io/crowdin-cli/commands/crowdin-bundle) and [`distribution`](https://crowdin.github.io/crowdin-cli/commands/crowdin-distribution) commands.
